### ISSUE TITLE
researcher pays too much prevents staking

### ISSUE THREAD

**Issue by Quezacoatl1** [gridcoin-community/Gridcoin-Research#1201]
JoeWinker and I experienced that @jamescowens already had a deeper look at, so maybe he can provide more info.

I did not stake for months now https://gridcoinstats.eu/cpid/1a7c8f7bbe5a14b0c1e205af3bd2d3d5 . The problem seems to be that the client actually finds blocks

"mining-created" : 6,
"mining-accepted" : 6,
"mining-kernels-found" : 6,

but it does not really stake them. I see many "researcher pays too much" statements in the logs. 

My node (quoting jim):

> This line is the critical one, and is the same thing that is afflicting joe winkler...
> 06/14/2018 18:12:58 ERROR: ConnectBlock[ResearchAge] : Researchers Reward Pays too much : Interest 144.464516 and Research 2378.900000 and StakeReward 2523.364001, OUT_POR 1423.619910, with Out_Interest 144.464516 for CPID 1a7c8f7bbe5a14b0c1e205af3bd2d3d5
> 06/14/2018 18:12:58 ERROR: ReorganizeChain: ConnectBlock

on a foreign node (quoting jim again, it is jims node):

> 06/14/2018 18:11:59 {SBC} new best {615df2c808787a7e86caf05e80d9b0e9d73df7b1d2aea4ba645add964a05ea2b 1276206} ;
> 06/14/2018 18:11:59 {PB}: ACC;
> 06/14/2018 18:12:21  Received block 20b710a04612c1c33fd7c391d19c83722aa862c46c0de740a65d5453adaeea32;
> 06/14/2018 18:12:21 AcceptBlock: Proof Of Stake V8 9
> 
> 06/14/2018 18:12:21 {SBC} new best {20b710a04612c1c33fd7c391d19c83722aa862c46c0de740a65d5453adaeea32 1276207} ;
> 06/14/2018 18:12:21 {PB}: ACC;
> 06/14/2018 18:12:58  Received block 867212f4547f8e0f1e4a70ebf39e1a70c60a48bf3a9982dbf314cd84e61602c0;
> 06/14/2018 18:12:58 AcceptBlock: Proof Of Stake V8 9
> 
> 06/14/2018 18:12:58 ERROR: ConnectBlock[ResearchAge] : Researchers Reward Pays too much : Interest 144.464516 and Research 2378.900000 and StakeReward 2523.364001, OUT_POR 1423.619910, with Out_Interest 144.464516 for CPID 1a7c8f7bbe5a14b0c1e205af3bd2d3d5
> 06/14/2018 18:12:58 ERROR: ReorganizeChain: ConnectBlock 867212f4547f8e0f1e4a70ebf39e1a70c60a48bf3a9982dbf314cd84e61602c0 failed
> 06/14/2018 18:12:58 Previous block 20b710a04612c1c33fd7c391d19c83722aa862c46c0de740a65d5453adaeea32
> 
> 06/14/2018 18:12:58 InvalidChainFound: invalid block=867212f4547f8e0f1e4a  height=1276208  trust=3275838333972824660711873320762369734724920483558585417587761448943652835288  blocktrust=4413562943  date=06/14/2018 18:12:48
> 
> 06/14/2018 18:12:58 InvalidChainFound:  current best=20b710a04612c1c33fd7  height=1276207  trust=3275838333972824660711873320762369734724920483558585417587761448939239272345  blocktrust=3753750439  date=06/14/2018 18:12:16

Some quotes that may shed some light on the problem

jim:

> This is a version of the trapezoid problem, I believe... 

> both quezacoatl and joewinkler have similar profile for their magnitude. Essentially triangular with very high mags in the relatively distant past with moderate to low mags now. The only reason that we are not getting major forking on this is that the block is being rejected by other nodes for paying too much and/or reorged out, which is a good thing. What I don’t understand is why it thinks the reward such be that high? Doing a rough calculation by hand you get numbers similar to gridcoinstats.eu reward estimate for quezacoatl, which is a lot lower.

> Another clue on this is that the RSA (BoincRewardPending) in the wallet is also sky high... >2000, when it should be more like 800 or so...

a user who I hopefully may quote:

> the BoincRewardPending is from same function that puts reward in block
> i suspect this is issue of inconsistent Lifetime data

**Comment by esspam** [comment]
I seem to have the same problem.
Not staked for over 2 Months even though the miner found blocks which got accepted (only by my own node, probably). (https://gridcoinstats.eu/cpid/91eccec4234172a96844856bd5ff8445)
I locally added a debug-command, which calculates the _BoincRewardPending_ for a given cpid (to find out if it is caused by inconsistent data) and it gives me something like 7k grc for your cpid - so I see no reason why my node should reject your block.
Oh, and I tried v3.7.12, as this was the last version that I staked a block with, but the pending boinc reward is the same as in v3.7.15. So to me it seems, it is not an error introduced in the latest versions.
However I couldn't really find out much more. Enabled some debug-output on my node but there is nothing about too high research rewards in the last 24 hrs. Also I don't know too much about the gridcoin-code.
Hopefully someone can help.
